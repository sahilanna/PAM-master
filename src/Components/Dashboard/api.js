import axios from "axios";
import { ngrokUrl } from "../../Assets/config";

let token = null;
const getAccessToken = () => {
  if(!token){
  const auth = JSON.parse(sessionStorage.getItem("item"));
  token = auth.token;
  console.log("token",token)
  }
  return token;
};


const api = axios.create({
  baseURL:ngrokUrl,
  headers: {
    "Content-Type": "application/json",
  },
});


api.interceptors.request.use(
  (config) => {
    const token = getAccessToken();
    if (token) {
      config.headers["AccessToken"] = token;
      config.headers["ngrok-skip-browser-warning"]="true"
    }
    return config;
  },
  (error) => {
    return Promise.reject(error);
  }
);

api.interceptors.response.use(
  (res) => {
    return res;
  },
  async (err) => {
    const originalConfig = err.config;
    if (err.response) {
      // Access Token was expired
      if (err.response.status === 401 && !originalConfig._retry) {
        originalConfig._retry = true;
        try {
          const token=getAccessToken()
          window.localStorage.setItem("AccessToken", token);
          api.defaults.headers.common["AccessToken"] = token;
          return api(originalConfig);
        } catch (_error) {
          if (_error.response && _error.response.data) {
            return Promise.reject(_error.response.data);
          }
          return Promise.reject(_error);
        }
      }
      if (err.response.status === 403 && err.response.data) {
        return Promise.reject(err.response.data);
      }
    }
    return Promise.reject(err);
  }
);
export default api;




