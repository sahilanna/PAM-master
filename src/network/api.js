import axios from "axios";
import { NGROK_URL } from "./config";
import { Navigate } from "react-router-dom";
let token = null;
const getAccessToken = () => {
  if (!token) {
    const auth = JSON.parse(sessionStorage.getItem("item"));
    token = auth.token;
  }
  return token;
};

const api = axios.create({
  baseURL: NGROK_URL,
  headers: {
    "Content-Type": "application/json",
  },
});

api.interceptors.request.use(
  (config) => {
    const token = getAccessToken();
    if (token) {
      config.headers["AccessToken"] = token;
      config.headers["ngrok-skip-browser-warning"] = "true";
    }
    return config;
  },
  (error) => {
    return Promise.reject(error);
  }
);

api.interceptors.response.use(
  (res) => {
    return res;
  },
  async (err) => {
    if (err.response) {
      if (err.response.status === 401) {
        Navigate("/Login");
      } else if (err.response.status === 403 && err.response.data) {
        return Promise.reject(err.response.data);
      }
    }
    return Promise.reject(err);
  }
);
export default api;
