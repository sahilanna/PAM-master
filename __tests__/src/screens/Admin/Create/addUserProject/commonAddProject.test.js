import React from 'react';
import { render, screen, } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { useNavigate, useLocation } from 'react-router-dom';
import CommonAddProject from '../../../../../../src/screens/Dashboard/Admin/Create/addUserProject/commonAddProject';
import '@testing-library/jest-dom';
import api from '../../../../../../src/network/api';



jest.mock('react-router-dom', () => ({
  useNavigate: jest.fn(),
  useLocation: jest.fn(),
}));

jest.mock('../../../../../../src/network/api', () => ({
  get: jest.fn(),
  post: jest.fn(),
  put: jest.fn(),
}));

describe('CommonAddProject', () => {
  const mockNavigate = jest.fn();
  const mockLocation = {
    state: {
      projectId: '123',
      projectName: 'Test Project',
    },
  };

  beforeEach(() => {
    useNavigate.mockReturnValue(mockNavigate);
    useLocation.mockReturnValue(mockLocation);

    jest.clearAllMocks();
  });

  it('renders the component for user role', async () => {
    const { container } = render(<CommonAddProject role="user" />);
    expect(container).toMatchSnapshot();
    
  });

  it('renders the component for pm role', async () => {
    const { container } = render(<CommonAddProject role="pm" />);
    expect(container).toMatchSnapshot();
    
  });

  it('handles user selection and OTP submission', async () => {
    // Mocking the API responses
    api.get.mockResolvedValueOnce({ data: [{ id: 1, name: 'User 1' }, { id: 2, name: 'User 2' }] });
    api.post.mockResolvedValueOnce({});
    api.post.mockResolvedValueOnce({ status: 200 });
    api.put.mockResolvedValueOnce({});

    render(<CommonAddProject role="user" />);


   
    const otpInput = screen.getByRole('textbox'); 
    userEvent.type(otpInput, '123456');
   
    
  });


  it('handles form submission without user selection', async () => {
    render(<CommonAddProject role="user" />);

   
    const submitButton = screen.getByText('Submit');
    userEvent.click(submitButton);

    
    expect(api.post).not.toHaveBeenCalled();
  });
});
