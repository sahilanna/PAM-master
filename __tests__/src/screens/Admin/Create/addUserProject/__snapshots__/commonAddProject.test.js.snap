// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonAddProject renders the component for pm role 1`] = `<div />`;

exports[`CommonAddProject renders the component for user role 1`] = `<div />`;
